### ISSUE TITLE
[Bilibili] Bangumi site changed; ERROR BiliBiliBangumi Unable to extract play info

### ISSUE THREAD

**Issue by aliuq** [yt-dlp/yt-dlp#6701]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a bug unrelated to a specific site
- [X] I've verified that I'm running yt-dlp version **2023.03.04** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Provide a description that is worded well enough to be understood

While trying to download Bilibili videos, I encountered an error that took me a long time to debug. I found that the Bangumi page structure had changed and there was no longer a window.__INITIAL_STATE__ field. Instead, an object was wrapped in a script tag, but the original fields were missing. It seems that API calls might be required to obtain the necessary data, but I attempted this and the results were incomplete.

```bash
[ 18:07:15 ] ❯ yt-dlp 'https://www.bilibili.com/bangumi/play/ep183876' -v
[debug] Command-line config: ['https://www.bilibili.com/bangumi/play/ep183876', '-v']
[debug] Portable config "C:\scoop_soft\apps\yt-dlp\current\yt-dlp.conf": []
[debug] Encodings: locale cp65001, fs utf-8, pref cp65001, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.03.04 [392389b7d] (win_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.25330-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg 6.0-full_build-www.gyan.dev (setts), ffprobe 6.0-full_build-www.gyan.dev
[debug] Optional libraries: Cryptodome-3.17, brotli-1.0.9, certifi-2022.12.07, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Loaded 1786 extractors
[BiliBiliBangumi] Extracting URL: https://www.bilibili.com/bangumi/play/ep183876
[BiliBiliBangumi] ep183876: Downloading webpage
ERROR: [BiliBiliBangumi] ep183876: Unable to extract play info; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "yt_dlp\extractor\common.py", line 694, in extract
  File "yt_dlp\extractor\bilibili.py", line 376, in _real_extract
  File "yt_dlp\extractor\common.py", line 1256, in _search_json
  File "yt_dlp\extractor\common.py", line 1242, in _search_regex
```

```python
_start_pattern = r'\<script\s*id=\"__NEXT_DATA__\"\s*type="application\/json\"\>'
_end_pattern = r'\<\/script\>'
json_data = self._search_json(_start_pattern, webpage, 'play info', video_id, end_pattern=_end_pattern)

epMap = traverse_obj(json_data, ('props', 'pageProps', 'dehydratedState', 'queries', 0, 'state', 'data', 'epMap'))
current_data = epMap[video_id.split('ep')[-1]]

aid = current_data['aid']
cid = current_data['cid']

download_json = self._download_json(f'https://api.bilibili.com/x/player/playurl?avid={aid}&cid={cid}&qn=80', video_id=video_id)
```

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', 'https://www.bilibili.com/bangumi/play/ep183876']
[debug] Portable config "C:\scoop_soft\apps\yt-dlp\current\yt-dlp.conf": []
[debug] Encodings: locale cp65001, fs utf-8, pref cp65001, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.03.04 [392389b7d] (win_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.25330-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg 6.0-full_build-www.gyan.dev (setts), ffprobe 6.0-full_build-www.gyan.dev
[debug] Optional libraries: Cryptodome-3.17, brotli-1.0.9, certifi-2022.12.07, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Loaded 1786 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.03.04, Current version: stable@2023.03.04
Current Build Hash: 5590c57bd0433ed239a2deaaf92e2ad6f37fe50f53664c821575cafe106a9421
yt-dlp is up to date (stable@2023.03.04)
[BiliBiliBangumi] Extracting URL: https://www.bilibili.com/bangumi/play/ep183876
[BiliBiliBangumi] ep183876: Downloading webpage
ERROR: [BiliBiliBangumi] ep183876: Unable to extract play info; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "yt_dlp\extractor\common.py", line 694, in extract
  File "yt_dlp\extractor\bilibili.py", line 376, in _real_extract
  File "yt_dlp\extractor\common.py", line 1256, in _search_json
  File "yt_dlp\extractor\common.py", line 1242, in _search_regex
```

**Comment by aliuq** [comment]
I noticed that the result is different in the browser and in the code. In `requests.get()` there is a `window.__playinfo__`, but it doesn't exist in the browser.

**Comment by hengyu95** [comment]
Same here, it doesn't work anymore.

I can manually find the video and audio requests in Dev Tools so it's definitely doable and not protected by some DRM or anything like that

**Comment by catscarlet** [comment]
Got same issue here.

Tried `--user-agent "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/113.0" --cookies-from-browser firefox --add-header "Referer:https://www.bilibili.com/"` but none of them worked.

**Comment by 20NE** [comment]
same

**Comment by GD-Slime** [comment]
Fixed in #7337. Waiting for merging.
